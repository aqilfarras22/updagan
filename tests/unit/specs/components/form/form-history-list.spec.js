import { shallowMount, createLocalVue } from '@vue/test-utils';

import formHistoryListComponent from '@/components/form/form-history-list';

const { it, describe, expect } = global;

const testPropsData = {
  formVersion: {
    history: [{ id: 1 }],
  },
};

const localVue = createLocalVue();

describe('components/form/form-history-list.vue', () => {
  const createMountedWrapper = (options = {}) => shallowMount(formHistoryListComponent, {
    localVue,
    propsData: { ...testPropsData },
    stubs: {
      TableComponent: {
        name: 'TableComponent',
        render: h => h('b', 'TableComponent'),
      },
    },
    ...options,
  });
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should have described child components', () => {
    expect(formHistoryListComponent.components.TableComponent).toBeTruthy();
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.fakeProp).toBeUndefined();
    expect(wrapper.vm.history).toEqual(testPropsData.history);
  });

  it('should have all required initial data', () => {
    const wrapper = createMountedWrapper();
    // initial data
    expect(wrapper.vm.approvalStatusText).toMatchSnapshot();
    expect(wrapper.vm.verificationStatusText).toMatchSnapshot();
    expect(wrapper.vm.historyStatusText).toMatchSnapshot();
    expect(wrapper.vm.historyStatus).toMatchSnapshot();
    expect(wrapper.vm.formHistoryFields).toMatchSnapshot();
  });

  it('should have method which emit "view" event: view(item)', () => {
    const wrapper = createMountedWrapper();
    wrapper.vm.view('item');
    // assert event has been emitted
    expect(wrapper.emitted().view).toBeTruthy();
    // assert event count
    expect(wrapper.emitted().view.length).toBe(1);
    // assert event payload
    expect(wrapper.emitted().view[0][0]).toBe('item');
  });

  it('has the expected html structure', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
