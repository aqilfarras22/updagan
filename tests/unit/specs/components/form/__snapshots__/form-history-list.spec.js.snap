// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-history-list.vue has the expected html structure 1`] = `
<b
  fields="[object Object],[object Object],[object Object],[object Object]"
  formversion="[object Object]"
  nogutters=""
  showempty=""
>
  TableComponent
</b>
`;

exports[`components/form/form-history-list.vue should have all required initial data 1`] = `
Object {
  "APPROVED": "Approved",
  "NOT_APPROVED": "Unapproved",
}
`;

exports[`components/form/form-history-list.vue should have all required initial data 2`] = `
Object {
  "NOT_VERIFIED": "Unverified",
  "VERIFIED": "Verified",
}
`;

exports[`components/form/form-history-list.vue should have all required initial data 3`] = `
Object {
  "MODIFIED": "Form updated",
}
`;

exports[`components/form/form-history-list.vue should have all required initial data 4`] = `
Object {
  "MODIFIED": "MODIFIED",
}
`;

exports[`components/form/form-history-list.vue should have all required initial data 5`] = `
Array [
  Object {
    "key": "createdOn",
    "label": "Date",
    "sortable": true,
  },
  Object {
    "key": "eventType",
    "label": "Action",
    "sortable": true,
  },
  Object {
    "key": "issuer",
    "label": "By",
    "sortable": true,
  },
  Object {
    "key": "view",
    "label": " ",
  },
]
`;
